### ISSUE TITLE
Cannot load data/interface/portrait/UNKNOWN_p01.jpg

### ISSUE THREAD

**Issue by michaliskambi** [eugeneloza/decoherence#1]
Running the latest "Decoherence" code on Linux 64-bit (x64_64) using the latest Castle Game Engine 6.3 (from GitHub), I get an error:

```
Exception "EFOpenError" :
Unable to open file "/home/michalis/sources/castle-engine/decoherence/data/interface/portrait/UNKNOWN_p01.jpg"

  $000000000046F499
  $000000000046F293
  $000000000079550E  DOWNLOAD,  line 417 of ../castle-engine/src/net/castledownload.pas
  $00000000005AAC62  LOADENCODEDIMAGE,  line 4225 of ../castle-engine/src/images/castleimages.pas
  $00000000005AB046  LOADENCODEDIMAGE,  line 4251 of ../castle-engine/src/images/castleimages.pas
  $00000000005AB3C3  LOADIMAGE,  line 4294 of ../castle-engine/src/images/castleimages.pas
  $0000000000BF6394  LOADIMAGESAFE,  line 74 of decoinputoutput.pas
  $0000000000BEBC01  INITCOMPOSITEINTERFACE,  line 215 of decointerfacecomposite.pas
  $0000000000BDC24B  INITINTERFACE,  line 336 of decointerface.pas
  $0000000000466969  LOADANDINITDATA,  line 228 of decoherence.pas
  $0000000000466BF5  APPLICATIONINITIALIZE,  line 290 of decoherence.pas
  $00000000004B69E8  CASTLEENGINEINITIALIZE,  line 4629 of ../castle-engine/src/window/castlewindow.pas
  $00000000004B0D5C  OPENCORE,  line 3171 of ../castle-engine/src/window/castlewindow.pas
  $00000000004B11C1  OPEN,  line 3209 of ../castle-engine/src/window/castlewindow.pas
  $00000000004B1362  OPEN,  line 3241 of ../castle-engine/src/window/castlewindow.pas
  $00000000004B32CC  OPENANDRUN,  line 3764 of ../castle-engine/src/window/castlewindow.pas
  $000000000040BB3B  main,  line 21 of decoherence_desktop.lpr
```

As far as I see, there is no file named `UNKNOWN*` (with any case, upper or lower) in the repository, so the referenced file is just missing from the repository.

(Or do I need to run some utility to generate it? I have not found a mention of it in the README.)

I have not looked too deep in the code -- if you think this is caused by some Castle Game Engine bug, or it's something specific to Linux (it may be -- on a standard Linux filesystems, filenames are case-sensitive), then please let me know, I will debug it more:)

Subsequently, the loading seems to take forever, I look at the loading animation forever (I assume it's caused by the above bug).

**Comment by eugeneloza** [comment]
Aww... yes, it's the problem, because UNKNOWN_p*.jpg are placeholders of unclear license (and likely to be proprietary), so I didn't upload them to github (they are in gitignore). Thou I should have made load process "fail-proof". I'll send you the files by e-mail.

**Comment by eugeneloza** [comment]
And another thing: you have to click to continue after loading has been complete. I'm working on showing the progressbar. Loading is usually ~2 sec. in release mode and ~15 sec in debug mode.

**Comment by michaliskambi** [comment]
I see. The portraits you send me indeed look very nice... A pity that they cannot be used freely.

I searched for a bit https://opengameart.org/ looking for portraits, but it's not easy to find a set of matching (in style) portraits, and fitting Wizardry-like mix of sci-fi/fantasy. I found https://opengameart.org/content/flare-portrait-pack-resized and https://opengameart.org/content/justin-nichols-sci-fi-portraits-png-psd-transparency , but these are only humans (16 fantasy portraits, 5 sci-fi portraits).

The loading goes fine now. BTW, looks like instead of clicking, I can also Alt+Tab to a different application, and then get back.

I submitted a simple pull request to load a "placeholder.png" in case the original "UNKNOWN_xxx" cannot be loaded, https://github.com/eugeneloza/decoherence/pull/2 . If you decide to take it, you can probably close this issue too :)

Regards!

**Comment by eugeneloza** [comment]
Yes, thanks a lot! I'm still trying to put myself together after Ph.D. defense... Nearly zero progress in last 3 weeks :(
I'll leave this issue open, because replacing portraits must be done in some future. The main idea is that there will be no human portraits (all characters are aliens, like in UNKNOWN_* files).
Yes, the loadscreen (should) stop after any keypress or mouse click / touch event.

**Comment by eugeneloza** [comment]
CC-BY Portraits ready, actually placeholders, but will do for some nearest future (and maybe even for mid-Alpha). I will upload them soon.
